### PR TITLE
chore: update lima example to use --with-terraform arg

### PR DESCRIPTION
**Pull request by johnstcn**
https://github.com/coder/coder/pull/5586 added the capability for `install.sh` to download and install Terraform automatically. Using this now in the example Lima specification.
Also no longer hard-coding the instance name in favour of `{{.Instance.Name}}` in the output that gets emitted upon successful instance provisioning.

### CODE REVIEW

File: examples/lima/coder.yaml [hunk 1]
```diff
@@ -65,25 +65,16 @@ provision:
...
-    script: |
-      #!/bin/bash
-      set -eux -o pipefail
-      command -v terraform >/dev/null 2>&1 && exit 0
-      DEBIAN_FRONTEND=noninteractive apt-get install -qqy unzip
-      rm -fv /tmp/terraform.zip || true
-      wget -qO /tmp/terraform.zip "https://releases.hashicorp.com/terraform/1.3.0/terraform_1.3.0_linux_$(dpkg --print-architecture).zip"
-      unzip /tmp/terraform.zip -d /usr/local/bin/
-      chmod +x /usr/local/bin/terraform
-      rm -fv /tmp/terraform.zip || true
```
Comment (johnstcn) [inline examples/lima/coder.yaml:78]: review: no longer required

File: examples/lima/coder.yaml [hunk 2]
```diff
@@ -65,25 +65,16 @@ provision:
...
   - mode: system
     script: |
       #!/bin/bash
       set -eux -o pipefail
       command -v coder >/dev/null 2>&1 && exit 0
       export DEBIAN_FRONTEND=noninteractive
       export HOME=/root
-      curl -fsSL https://coder.com/install.sh | sh
+      # Install some dependencies
+      apt-get install -qqy unzip
```
Comment (johnstcn) [inline examples/lima/coder.yaml:76]: review: required for `--with-terraform`

File: examples/lima/coder.yaml [hunk 3]
```diff
@@ -93,10 +84,8 @@ provision:
       # Ensure coder starts on boot
       systemctl enable coder
       systemctl start coder
-      # Wait for Coder to have downloaded Terraform
-      timeout 60s bash -c 'until /var/cache/coder/terraform version >/dev/null 2>&1; do sleep 1; done'
-      # Coder restarts after downloading Terraform, wait for it to become available
-      timeout 60s bash -c 'until nc -z localhost 3000 > /dev/null 2>&1; do sleep 1; done'
+      # Wait for Terraform to be installed
+      timeout 60s bash -c 'until /usr/local/bin/terraform version >/dev/null 2>&1; do sleep 1; done'
```
Comment (johnstcn) [inline examples/lima/coder.yaml:88]: review: terraform path updated

File: examples/lima/coder.yaml [hunk 4]
```diff
@@ -137,14 +126,24 @@ probes:
...
+        exit 1
+      fi
+    hint: |
+      See "/var/log/cloud-init-output.log" in the guest.
 message: |
   All Done! Your Coder instance is accessible at http://localhost:3000
 
   Username: "admin@coder.com"
-  Password: Run `LIMA_INSTANCE=coder lima cat /home/${USER}.linux/.config/coderv2/password` 🤫
+  Password: Run `LIMA_INSTANCE={{.Instance.Name}} lima cat /home/${USER}.linux/.config/coderv2/password` 🤫
```
Comment (johnstcn) [inline examples/lima/coder.yaml:143]: review: see https://github.com/lima-vm/lima/blob/25b137d4c54a5a251fb7cca4adf1892097e1a6f2/pkg/store/instance.go#L36

**Review by mafredri — APPROVED** [review]
Nice! 👍🏻

**Review by mtojek — APPROVED** [review]

File: examples/lima/coder.yaml (hunk 3 above)
Comment (johnstcn) [reply examples/lima/coder.yaml:88]: My thinking here is that if terraform installation fails for some reason, it will be more obvious to the end-user.

**Pull request merged**